create_assets_json.py

```diff
@@ -1,22 +1,76 @@
 from copy import deepcopy
 import json
 import os
+import hashlib
 
+with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'assets.json'), 'r') as infile:
+	assets = json.load(infile)
 
-CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config.json'), 'r') as infile:
+	config = json.load(infile)
 
-with open(os.path.join(CURRENT_DIR, 'assets.json'), 'r') as infile:
-	assets = json.load(infile)
+def get_l1_network_config_from_fuel(networks, type, chain):
+	for network in networks:
+		if network.get('type') == type and network.get('chain') == chain:
+			return network
+
+def parse_address_to_bytes(address):
+    if address.startswith('0x'):
+        address = address[2:]  # Remove the '0x' prefix
+    return bytes.fromhex(address)
+
+def pad_address(address, byte_len):
+	hex_address = address.hex()
+	padded_hex_address = hex_address.zfill(byte_len)
+	return bytes.fromhex(padded_hex_address)
+
+def generate_fuel_network_config(asset, chain, network):
+	base_config = config['fuelAddressConfig'][chain]
+	l1_network = get_l1_network_config_from_fuel(asset['networks'], base_config['l1_type'], base_config['l1_chain'])
+	if l1_network is None:
+		del network
+		return
+
+	l1_chain_id = config['chainIds'][base_config['l1_type']][base_config['l1_chain']]
+	l1_contract_address = l1_network['address']
+
+	# set the contract ID
+	network['contractId'] = base_config['l2_contract']
+	sub_id = calculate_fuel_sub_id(l1_chain_id, l1_contract_address)
+	asset_id = calculate_fuel_asset_id(network['contractId'], sub_id)
+	network['subId'] = f'0x{sub_id.hexdigest()}'
+	network['assetId'] = f'0x{asset_id.hexdigest()}'
+
+	return
 
-with open(os.path.join(CURRENT_DIR, 'chains.json'), 'r') as infile:
-	chains = json.load(infile)
+def calculate_fuel_sub_id(chain_id, address):
+	# chain ID is always 1
+	# https://github.com/FuelLabs/fuel-bridge/blob/0b6e3b198707d9fd7000d255b0ad3164d7e64507/packages/fungible-token/bridge-fungible-token/implementation/src/main.sw#L496
+	chain_id = bytes('1'.encode('utf-8')) # number 1 in ASCII
+	token_id = bytes(32) # zeros, not used
 
-with open(os.path.join(CURRENT_DIR, 'cdn.json'), 'r') as infile:
-	cdn = json.load(infile)
+	# parse eth address and ensure padding
+	parsed_address = parse_address_to_bytes(address)
+	padded_address = pad_address(parsed_address, 64)
+
+	# setup bytes to sign
+	bytes_to_sign = bytearray(chain_id)
```

Comment:
rename bytes_to_sign to bytes_to_hash?

Reply:
Good point - I cleaned up the code a bit to refactor this into a `sha256` function.